src/main.rs

```diff
@@ -880,8 +881,26 @@ fn main() {
             path.map(|s| std::fs::remove_file(s).ok());
             0
         }
-        Err(e) => {
-            eprintln!("{e}");
+        Err(top_error) => {
+            eprint!("Error: ");
+            if top_error.source().is_none() {
+                eprintln!("Cloud Hypervisor exited with the following error:");
+                eprintln!("  {top_error}");
+            } else {
+                eprintln!("Cloud Hypervisor exited with the following chain of errors:");
+                eprintln!("  0: {top_error}");
+                let mut level = 1;
+                let mut next_error: &dyn StdError = &top_error;
+                while let Some(sub_error) = next_error.source() {
```

Comment:
nit: Maybe some combination of [`std::iter::successors`](https://doc.rust-lang.org/std/iter/fn.successors.html) and [`enumerate`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) can streamline this a bit.